### DISCUSSION TITLE
Ignore Named Entities

### DISCUSSION THREAD

**Discussion by noahberhe** [MaartenGr/BERTopic#1191] [Q&A]
Hello,

Do I need clean the data of names / places that occur frequently, or is there a way to tell BERTopic to ignore e.g. named entities or pronouns as potential topics?

On that note... can we ignore dates / numbers etc from being considered as topics?

I know PartOfSpeech exists, but I believe that just changes the presentation of a topic that's already been created based on a named entity?

Thank you 
Noah

**Comment by MaartenGr** [comment]
There is not one answer to this but it depends on your topic representations. If they contain too many entities of names and places, then it might be worthwhile to either clean them or use something like `MaximalMarginalRelevance` or `KeyBERTInspired`. `MaximalMarginalRelevance` takes the semantic similarity of words to topic embeddings into account. Since entities like names and places often contribute less to the semantic nature of sentences, at least that is what I typically experience with sentence-transformer embeddings, they get removed with `MaximalMarginalRelevance` or `KeyBERTInspired`. 

> On that note... can we ignore dates / numbers etc from being considered as topics?

You can remove those as part of the preprocessing step in the `CountVectorizer`. 

> I know PartOfSpeech exists, but I believe that just changes the presentation of a topic that's already been created based on a named entity?

Indeed, it takes a topic with candidate keywords and fine-tunes those based on whether those candidate keywords are part of your selected part-of-speech, like nouns.

**Reply by noahberhe** [reply]
But the MaximalMarginalRelevance or KeyBERTInspired solutions are a representation layer after the topic clusters have been created right? So wouldn't a topic based on occurrences of the term "Apple Inc" for example, just be re-represented in other terms?

Or are you saying because entities contribute less to the semantic nature of sentences, that they don't tend to define topics?

Thank you :)

**Reply by MaartenGr** [reply]
> Or are you saying because entities contribute less to the semantic nature of sentences, that they don't tend to define topics?

Yes, when you want to rerank candidate keywords and you are using a semantic-based method, like embedding models, they tend to underestimate the influence these entities have on the entire meaning of a sentence. As a result, these methods might *indirectly* remove those entities.

**Reply by MolkaSlama** [reply]
I have used Bertopic in a TikTok dataset and I get topics represented by "Sephora" and other brands.

**Reply by MaartenGr** [reply]
@MolkaSlama You could assign them as stop words through the [CountVectorizer](https://maartengr.github.io/BERTopic/getting_started/vectorizers/vectorizers.html#stop_words). You could remove them beforehand or even create a custom NER pipeline to remove them using a custom representation model but that would likely get a bit too complex and is often unnecessary.
